plugins/inputs/procstat/procstat.go

```diff
@@ -600,16 +608,8 @@ func (p *Procstat) winServicePIDs() ([]PID, error) {
 	return pids, nil
 }
 
-func (p *Procstat) Init() error {
-	if strings.ToLower(p.Mode) == "solaris" {
-		p.solarisMode = true
-	}
-
-	return nil
-}
-
 func init() {
 	inputs.Add("procstat", func() telegraf.Input {
-		return &Procstat{}
+		return &Procstat{createProcess: NewProc}
```

Comment:
There was a conditional logic here before, and I'm not sure if removing it will cause compatibility issues.
（ﾉ´д｀）
```
if p.createProcess == nil {
		p.createProcess = defaultProcess